velox/functions/lib/string/tests/StringImplTest.cpp

```diff
@@ -431,6 +431,17 @@ TEST_F(StringImplTest, getByteRange) {
     EXPECT_EQ(expectedStartByteIndex, range.first);
     EXPECT_EQ(expectedEndByteIndex, range.second);
   }
+
+  // Test bad unicode strings.
+  std::string badUnicode = "aa\xff ";
```

Comment:
should we add tests to TEST_F(StringFunctionsTest, rpad) ?

Also, would be good to understand what is the difference between these 2 cases, maybe add a one line comment?

Reply:
The first test exercises the first loop in getByteRange, and the second test exercises the second loop. The tests are engineered so that those loops get activated with bad unicode. I will add the comment.